api/src/lib.rs

```diff
@@ -584,30 +610,49 @@ impl Client {
             .json::<Response<EmptySuccess, ErrorT>>()
             .chain_err(|| ErrorKind::BadJsonResponse)?
             .into_result(status)
-            .map(|_| ())
+            .map_or_else(
+                // Ignore 404 not found if the request had to be re-tried - assume the target
+                // object was deleted on a previous incomplete request.
+                |e| {
```

Comment:
nit. As a matter of style we stay away from single-letter variables. It's easy for things to get out of hand pretty quickly even if the original code is pretty straight-forward.

So here for instance, `error` would be preferred.